sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -76,6 +76,24 @@ abstract class DataType {
     DataType.equalsIgnoreNullability(this, other)
 
   /**
+   * Check if `this` is a primitive data type.
+   */
+  def isPrimitive: Boolean = {
```

Comment:
Let's remove this. "primitive" is not well defined. You can just check whether the type is AtomicType where you need to call this.